src/python/WMCore/MicroService/MSPileup/MSPileupTasks.py

```diff
@@ -57,13 +63,28 @@ def pileupSizeTask(self):
             rucioAuthUrl = self.rucioClient.rucioParams['auth_host']
             rucioHostUrl = self.rucioClient.rucioParams['rucio_host']
             rucioToken, _tokenValidity = getRucioToken(rucioAuthUrl, self.rucioAccount)
-            containers = [r['pileupName'] for r in docs]
-            msg = f"Fetching pileup size for {len(containers)} containers "
+            rucioScope = self.rucioScope
+            cmsContainers = []
+            customContainers = []
+            for doc in docs:
+                if doc.get('customName', '') != '':
+                    customContainers.append(doc['customName'])
+                else:
+                    cmsContainers.append(doc['pileupName'])
+            msg = f"Fetching pileup size for {len(cmsContainers)} cms containers "
+            msg += f"and {len(customContainers)} custom containers "
             msg += f"against rucio url: {rucioHostUrl}"
             self.logger.info(msg)
-            datasetSizes = getPileupContainerSizesRucio(containers, rucioHostUrl, rucioToken)
+            if len(cmsContainers) > 0:
+                datasetSizes = getPileupContainerSizesRucio(cmsContainers, rucioHostUrl, rucioToken)
+            if len(customContainers) > 0:
+                rucioScope = spec.get('customRucioScope', 'group.wmcore')
+                datasetSizes = getPileupContainerSizesRucio(customContainers, rucioHostUrl, rucioToken, scope=rucioScope)
```

Comment:
and here you overwrite `datasetSizes`.  This needs to be fixed.

Reply:
you're right, I added extra code to cope up with dict return types., feel free to check latest code (already squashed).